### PR TITLE
[SYCL] Enable loading cl_intel_devicelib_assert fallback spv

### PR DESCRIPTION
**Pull request by jinge90**
Previously, we disabled cl_intel_devicelib_assert fallback spv in https://github.com/intel/llvm/pull/6176 to bypass a Clang FE issue which existed in legacy compiler.
Now, the legacy compiler is out of date, we don't need this workaround any more, so enable loading cl_intel_devicelib_assert fallback spv again. This PR also updates some comments in sycl-post-link.

### CODE REVIEW

**Review by AlexeySachkov — APPROVED** [review]
`sycl-post-link` change is non-functional, LGTM

File: llvm/tools/sycl-post-link/SYCLDeviceLibReqMask.cpp [hunk 1]
```diff
@@ -657,16 +657,16 @@ SYCLDeviceLibFuncMap SDLMap = {
...
-// fallback-cmath:        0x2
-// fallback-cmath-fp64:   0x4
-// fallback-complex:      0x8
-// fallback-complex-fp64: 0x10
-// fallback-cstring:      0x20
-// fallback-imf:          0x40
-// fallback-imf-fp64:     0x80
-// fallback-imf-bf16:     0x100
-// fallback-bfloat16:     0x200
+// cl_intel_devicelib_assert:        0x1
```
Comment (maksimsab) [inline llvm/tools/sycl-post-link/SYCLDeviceLibReqMask.cpp:660]: I am a bit concerned that none of these values are present in SDLMap above.

**Review by maarquitos14 — APPROVED** [review]
Runtime change look good to me.

File: llvm/tools/sycl-post-link/SYCLDeviceLibReqMask.cpp (hunk 1 above)
Comment (jinge90) [reply llvm/tools/sycl-post-link/SYCLDeviceLibReqMask.cpp:660]: > I am a bit concerned that none of these values are present in SDLMap above. @jinge90 Any comments on that?

Hi, @maksimsab 
Which "values" do you refer to?
Thanks very much.

Comment (maksimsab) [reply llvm/tools/sycl-post-link/SYCLDeviceLibReqMask.cpp:660]: Sorry, I've found it.

**Comment by jinge90** [comment]
Hi, @intel/llvm-gatekeepers 
Could you help merge this patch? AWS CUDA test failed but it looks like to be environment issue:

" Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error running hook #0: error running hook: exit status 1, stdout: , stderr: Auto-detected mode as 'legacy'
  nvidia-container-cli: requirement error: unsatisfied condition: cuda>=12.1, please update your driver to a newer version, or use an earlier cuda container: unknown
  Error: failed to start containers: cdf8a38d2343157c9867734eccc3c893063ba40d99fa4887681d2d1640b074bc
  Error: Docker start fail with exit code 1"
Hi, @tfzhu could you confirm whether the AWS CUDA failure is caused by environment error?
Thanks very much.

**Comment by tfzhu** [comment]
> Hi, @intel/llvm-gatekeepers Could you help merge this patch? AWS CUDA test failed but it looks like to be environment issue:
> 
> " Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error running hook #0: error running hook: exit status 1, stdout: , stderr: Auto-detected mode as 'legacy' nvidia-container-cli: requirement error: unsatisfied condition: cuda>=12.1, please update your driver to a newer version, or use an earlier cuda container: unknown Error: failed to start containers: cdf8a38d2343157c9867734eccc3c893063ba40d99fa4887681d2d1640b074bc Error: Docker start fail with exit code 1" Hi, @tfzhu could you confirm whether the AWS CUDA failure is caused by environment error? Thanks very much.

Need to check with @aelovikov-intel. Re-run also failed.

**Comment by jinge90** [comment]
Hi, @intel/llvm-gatekeepers 
Could you help merge this patch?
Thanks very much.

**Pull request merged**